qiskit/compiler/assemble.py

```diff
@@ -16,128 +16,136 @@
 import uuid
 import copy
 
-from qiskit.circuit import QuantumCircuit
+from typing import Union, List, Dict, Optional
+from qiskit.circuit import QuantumCircuit, Qubit
 from qiskit.exceptions import QiskitError
 from qiskit.pulse import ScheduleComponent, LoConfig
 from qiskit.assembler.run_config import RunConfig
 from qiskit.assembler import assemble_circuits, assemble_schedules
-from qiskit.qobj import QobjHeader
+from qiskit.qobj import QobjHeader, Qobj
 from qiskit.qobj.utils import MeasLevel, MeasReturnType
 from qiskit.validation.jsonschema import SchemaValidationError
+from qiskit.providers import BaseBackend
+from qiskit.pulse.channels import PulseChannel
 
 
 # TODO: parallelize over the experiments (serialize each separately, then add global header/config)
-def assemble(experiments,
-             backend=None,
-             qobj_id=None, qobj_header=None,
-             shots=None, memory=False, max_credits=None, seed_simulator=None,
-             qubit_lo_freq=None, meas_lo_freq=None,
-             qubit_lo_range=None, meas_lo_range=None,
-             schedule_los=None, meas_level=MeasLevel.CLASSIFIED,
-             meas_return=MeasReturnType.AVERAGE, meas_map=None,
-             memory_slot_size=100, rep_time=None, parameter_binds=None,
-             parametric_pulses=None,
-             **run_config):
+def assemble(experiments: Union[QuantumCircuit, List[QuantumCircuit]],
+             backend: Optional[BaseBackend] = None,
+             qobj_id: Optional[str] = None, qobj_header: Optional[Union[QobjHeader, Dict]] = None,
```

Comment:
For a long function signature like this, the args should each go on their own line